test/post.test.js

```diff
@@ -92,7 +92,34 @@ describe('Post data class tests', () => {
     expect(() => createPostWithFeed({})).toThrow();
     expect(() => createPostWithFeed(feed)).not.toThrow();
   });
-
+  test('Post constructor should throw if invalid dates is passed', () => {
+    const createPostWithDates = (datePublished, dateUpdated) =>
+      new Post(data.title, data.html, datePublished, dateUpdated, data.url, data.guid, feed);
+    expect(() =>
+      createPostWithDates(
+        new Date('Thu, 20 Nov 2014 18:59:18 UTC'),
```

Comment:
All of these tests are passing a `Date` instance, whether valid or invalid.  To hit that final `throw` that you're trying to test, you need to pass `null` or `undefined` or something that falls through those ifs.

Reply:
So by a valid date you mean is the date object null or not? Not something like Dec. 40 2018?

Reply:
Stay focused on what this issue is all about: you are trying to get that "throw" line to be hit in the tests.  You need a test case that causes it to happen.  To do that, you need to stop passing a Date object, or you will always fall into the existing `if` statements.

Reply:
Ohhh I think I'm understanding, when we said throw if the date is invalid I thought u meant to check for something like Sept 40 2020. Turns out the Date() object already does that so all we need to worry about is if a Date object is being passed. ill keep working on this! Thanks for the explanation, it really helped!
![image](https://user-images.githubusercontent.com/36198917/99154694-bbe54900-267f-11eb-95fe-666ba98c4921.png)

Reply:
If all you're doing is writing tests you can just checkout master, create a new branch and just pull the updated test file from this branch into your new branch. You'll have to change one of your tests though (the one that has to throw), you can pass it two integers and make it fail that way.

Reply:
Sorry just saw this, I fixed everything and pushed to the issue-1323 branch. Would you like me to make a new branch eg. issue-1323-fixed and make the pr for there? As of now everything is updated on the issue-1323 branch

Reply:
No worries, that's fine